libindy_vdr/src/resolver/did_document.rs

```diff
@@ -173,12 +174,52 @@ fn merge_diddoc(base: &mut SJsonValue, content: &SJsonValue) {
     }
 }
 
-fn expand_verkey(id: &str, verkey: &str) -> String {
-    if let Some(stripped) = verkey.strip_prefix('~') {
-        format!("{}{}", id, stripped)
+// Returns raw verkey in case of conversion errors, otherwise 'default' indy handling
+pub fn expand_verkey(id: &str, verkey: &str) -> String {
+    let expanded_verkey = expand_verkey_indy(id, verkey);
+    if expanded_verkey.is_err() {
+        return verkey.to_string();
+    }
+    return expanded_verkey.unwrap();
+}
+
+pub fn expand_verkey_indy(id: &str, verkey: &str) -> VdrResult<String> {
+    // separate optional crypto_type from verkey part
+    let (key, key_type) = if verkey.contains(":") {
+        let vec: Vec<&str> = verkey.split(":").collect();
+        match vec.len() {
+            0 | 1 => {
```

Comment:
You check here for a length of 0 or 1, but you never check the total element count. In the match case below `vec[vec.len() - 1]` is called which might be the 99th element. I think a length check here would be a good addition.

Reply:
I will change that whole key_type part to expect ecactly 2 outputs and error on anything else - the indy-sdk implementation kinda expects the same.